src/internal/obj/local_client.go

```diff
@@ -157,27 +154,27 @@ func (c *fsClient) transformError(err error, name string) error {
 	return err
 }
 
-func (c *fsClient) closeFile(retErr *error, f *os.File) {
+func (c *fsClient) closeFile(ctx context.Context, retErr *error, f *os.File) {
 	err := f.Close()
```

Comment:
It would be ideal to remove this complicated logic and:
```go
if err := f.Close(); err != nil {
   if !strings.Contains(err, "already closed") {
       errors.JoinInto(retErr, errors.Wrap(err, "close"))
   }
}
```